model/project_vars.go

```diff
@@ -335,17 +335,37 @@ const defaultParameterStoreAccessTimeout = 30 * time.Second
 func (projectVars *ProjectVars) Upsert() (*adb.ChangeInfo, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultParameterStoreAccessTimeout)
 	defer cancel()
-	isPSEnabled, err := isParameterStoreEnabledForProject(ctx, projectVars.Id)
+
+	ref, isRepoRef, err := projectVars.findProjectRef()
 	grip.Error(message.WrapError(err, message.Fields{
-		"message":    "could not check if Parameter Store was enabled for project; assuming it's disabled and falling back to using the DB",
+		"message":    "could not get project ref to check if Parameter Store is enabled for project; assuming it's disabled and falling back to using the DB",
 		"project_id": projectVars.Id,
+		"epic":       "DEVPROD-5552",
 	}))
-	if isPSEnabled {
-		grip.Error(message.WrapError(projectVars.upsertParameterStore(ctx), message.Fields{
-			"message":    "could not upsert project vars into Parameter Store; falling back to using the DB",
+	var isPSEnabled bool
+	if ref != nil {
+		isPSEnabled, err = isParameterStoreEnabledForProject(ctx, ref)
+		grip.Error(message.WrapError(err, message.Fields{
+			"message":    "could not check if Parameter Store is enabled for project; assuming it's disabled and falling back to using the DB",
 			"project_id": projectVars.Id,
+			"epic":       "DEVPROD-5552",
 		}))
 	}
+	if isPSEnabled {
+		if !ref.ParameterStoreVarsSynced {
+			grip.Error(message.WrapError(fullSyncToParameterStore(ctx, projectVars, ref, isRepoRef), message.Fields{
+				"message":    "could not upsert project vars into Parameter Store; falling back to using the DB",
+				"project_id": projectVars.Id,
+				"epic":       "DEVPROD-5552",
+			}))
+		} else {
+			grip.Error(message.WrapError(projectVars.upsertParameterStore(ctx), message.Fields{
+				"message":    "could not upsert project vars into Parameter Store; falling back to using the DB",
+				"project_id": projectVars.Id,
+				"epic":       "DEVPROD-5552",
+			}))
```

Comment:
since message.Fields is the same can we save it in a variable and reuse it to avoid duplication?

Reply:
I guess we could, but I don't see a significant benefit to consolidating the logic since the continue on error will go away soon anyways.